File: single-language-analysis/dataset-files/java/TestSmell007.java
// Original URL: https://github.com/noel-austin/CSU33012-Calculator/blob/9dee51c9af4d9ac7f8a161acb5baeb2ff1dd6691/MainTest.java#L32-L46

public class TestSmell007 {
	@Test
	public void testSubtraction() {
		String result = Main.toDoTheOperation(Main.recognize("10-10-10-10"));
		assertEquals("-20.0", result);
		result = Main.toDoTheOperation(Main.recognize("0-0"));
		assertEquals("0.0", result);
		result = Main.toDoTheOperation(Main.recognize("10-10-10-20"));
		assertEquals("-30.0", result);
		result = Main.toDoTheOperation(Main.recognize("1-1"));
		assertEquals("0.0", result);
		result = Main.toDoTheOperation(Main.recognize("10-10-10-10-10-10-10-10-10-10-10-10"));
		assertEquals("-100.0", result);
		result = Main.toDoTheOperation(Main.recognize("0-25-25-25"));
		assertEquals("-75.0", result);
	}
}

File: single-language-analysis/dataset-files/python/test_smell_007.py
import pytest

class TestMainTest:
    def test_smell_007(self):
        result = Main.to_do_the_operation(Main.recognize("10-10-10-10"))
        assert result == "-20.0"
        result = Main.to_do_the_operation(Main.recognize("0-0"))
        assert result == "0.0"
        result = Main.to_do_the_operation(Main.recognize("10-10-10-20"))
        assert result == "-30.0"
        result = Main.to_do_the_operation(Main.recognize("1-1"))
        assert result == "0.0"
        result = Main.to_do_the_operation(Main.recognize("10-10-10-10-10-10-10-10-10-10-10-10"))
        assert result == "-100.0"
        result = Main.to_do_the_operation(Main.recognize("0-25-25-25"))
        assert result == "-75.0"